File: leetcode3/최원준/3622. Check Divisibility by Digit Sum and Product.py
#

'''
1. 아이디어 :
-

2. 시간복잡도 :
    O(len(n))

3. 자료구조/알고리즘 :
-

'''

class Solution:
    def checkDivisibility(self, n: int) -> bool:
        
        string = str(n)
        n_sum = 0
        n_product = 1

        for c in string:
            n_int = int(c)
            n_sum += n_int
            n_product *= n_int
        
        return n%(n_sum+n_product) == 0

File: leetcode3/염혜정/3622. Check Divisibility by Digit Sum and Product.java
// n을 각 자릿수를 더한 sum / 곱한 값인 product를 더한 값으로 나눠준다.
// O(log n)

class Solution {
    public boolean checkDivisibility(int n) {
        int sum = 0;
        int product = 1;
        int temp = n;

        while (temp>0) {
            int digit = temp % 10;
            sum += digit;
            product *= digit;
            temp /= 10;
        }

        boolean result = n % (sum+product) == 0? true: false;
        return result;
    }
}
